influxql/iterator.gen.go

```diff
@@ -956,6 +956,90 @@ func (itr *floatReaderIterator) Next() *FloatPoint {
 	return p
 }
 
+// floatMeanIterator returns the mean of values for every interval.
+type floatMeanIterator struct {
+	input  *bufFloatIterator
+	opt    IteratorOptions
+	points []*FloatPoint
+}
+
+// Close closes the iterator and all child iterators.
+func (itr *floatMeanIterator) Close() error { return itr.input.Close() }
+
+// Next returns the next value.
+func (itr *floatMeanIterator) Next() *FloatPoint {
+	// Calculate the next window if we have no more points.
+	if len(itr.points) == 0 {
+		itr.points = itr.next()
+		if len(itr.points) == 0 {
+			return nil
+		}
+	}
+
+	// Pop next point off the stack.
+	p := itr.points[len(itr.points)-1]
+	itr.points = itr.points[:len(itr.points)-1]
+	return p
+}
+
+func (itr *floatMeanIterator) next() []*FloatPoint {
+	startTime, endTime := itr.opt.Window(itr.input.peekTime())
+
+	// Create points by tags.
+	m := make(map[string]struct {
+		Name  string
+		Tags  Tags
+		Sum   float64
+		Count uint32
+	})
+	for {
+		// Read next point.
+		curr := itr.input.NextInWindow(startTime, endTime)
+		if curr == nil {
+			break
+		} else if curr.Nil {
+			continue
+		}
+		tags := curr.Tags.Subset(itr.opt.Dimensions)
+		id := curr.Name + "\x00" + tags.ID()
+
+		prev, ok := m[id]
+		if !ok {
+			prev.Name = curr.Name
+			prev.Tags = tags
+		}
+
+		if curr.Aggregated != 0 {
+			prev.Sum += curr.Value * float64(curr.Aggregated)
+			prev.Count += curr.Aggregated
+		} else {
+			prev.Sum += curr.Value
+			prev.Count++
+		}
```

Comment:
Ah, I think I can do something about this. I can try making an interface that could be used for aggregating and reducing the result. I can try mocking out an example and putting it in a gist so you can evaluate if it fits your needs. That ok?